### PR TITLE
Allow redis clients other than the specific pure perl 'Redis' module.…

### PR DESCRIPTION
**Pull request by djstevenson**
… References issue #17

### CODE REVIEW

**Comment by djstevenson** [comment]
Please note that this change adds a test-time dependency on Test::Exception

File: lib/Resque.pm [hunk 1]
```diff
@@ -97,7 +101,7 @@ has redis => (
     is      => 'ro',
     lazy    => 1,
     coerce  => 1,
-    isa     => 'Sugar::Redis',
+    isa     => 'Sugar::Redis|Object',
```
Comment (oschwald) [inline lib/Resque.pm]: I wonder if [duck_type](https://metacpan.org/pod/Moose::Util::TypeConstraints#duck_type-name-methods) would make more sense for this type.

Comment (djstevenson) [inline lib/Resque.pm]: I agree with you!

I must admit, I wasn't familiar with duck_type.  But I did think that having "something|Object"  does look a bit odd!

I'll try it duck_type, and upload an update assuming it works out OK.

Comment (djstevenson) [inline lib/Resque.pm]: Ah, I see that the standard pure perl Redis module doesn't actually implement the Redis commands directly, only via AUTOLOAD, so 'can' doesn't really work, so duck_type is going to be tricky, and not worth the effort of hacking in, after all.

Cheers!

**Pull request merged**